internal/backend/remote-state/s3/backend_state.go

```diff
@@ -29,25 +30,33 @@ func (b *Backend) Workspaces() ([]string, error) {
 		prefix = b.workspaceKeyPrefix + "/"
 	}
 
-	params := &s3.ListObjectsInput{
-		Bucket:  &b.bucketName,
+	params := &s3.ListObjectsV2Input{
+		Bucket:  aws.String(b.bucketName),
 		Prefix:  aws.String(prefix),
-		MaxKeys: aws.Int64(maxKeys),
+		MaxKeys: maxKeys,
 	}
 
 	wss := []string{backend.DefaultStateName}
-	err := b.s3Client.ListObjectsPages(params, func(page *s3.ListObjectsOutput, lastPage bool) bool {
+	pg := s3.NewListObjectsV2Paginator(b.s3Client, params)
+
+	ctx := context.TODO()
```

Comment:
Nit: wouldn't that normally be `context.Background()`

Reply:
eeeh, ideally you would pass the context, but since it breaks the interface and requires a lot of unrelated changes, I went with `context.TODO()` with the idea that we'll do that someday.

Reply:
Issue created to fix it in the future: https://github.com/opentofu/opentofu/issues/753